addons/account/models/account_journal.py

```diff
@@ -721,6 +697,14 @@ def _get_journal_bank_account_balance(self, domain=None):
         nb_lines, balance, amount_currency = self._cr.fetchone()
         return amount_currency if journal_currency else balance, nb_lines
 
+    def _get_journal_inbound_outstanding_payment_accounts(self):
+        return self.inbound_payment_method_line_ids.mapped('payment_account_id') \
+                   + self.company_id.account_journal_payment_debit_account_id
```

Comment:
Missing `self.ensure_one()`
You should not fallback on the company if not needed.
```
need_fallback = False
for line in self.inbound_payment_method_line_ids:
     account_ids.add(line.payment_account_id.id or self.company_id.account_journal_payment_debit_account_id.id)        
return self.env['account.account'].browse(account_ids)
```
Note: If `account_journal_payment_debit_account_id` is not required, don't add `False` into `account_ids`

Reply:
Note: Are these methods still necessary?